library/common/engine.h

```diff
@@ -21,11 +21,25 @@ namespace Envoy {
 
 class Engine {
 public:
+  /**
+   * Constructor for a new engine instance.
+   * @param callbacks, the callbacks to be used for streams started through the engine.
+   * @param config, the Envoy configuration to use when starting the instance.
+   * @param log_level, the log level with which to configure the engine.
+   * @param preferred_network, the preferred network that should be used for new streams.
```

Comment:
This is a dynamic value that is updated based on OS hooks. This is simple a handle to retrieve the current value.

Reply:
ah yeah good catch